crypto/src/oxend_key.cpp

```diff
@@ -57,4 +60,34 @@ x25519_pubkey x25519_seckey::pubkey() const {
     return pk;
 };
 
+template <typename T>
+static T parse_pubkey(std::string_view pubkey_in) {
```

Comment:
return `std::optional<T>` ?

Reply:
Not necessary for the pubkey types: they have a built-in `operator bool()` that returns false if set to an all-0 value.